crates/libs/core/src/types/client/partition.rs

```diff
@@ -233,4 +234,25 @@ impl PartitionLoadInformation {
             secondary_load_metric_reports,
         }
     }
+
+    /// Returns the partition id
+    pub fn partition_id(&self) -> GUID {
+        self.partition_id
+    }
+
+    /// Returns a list of load metric reports for primary replica
+    pub fn primary_load_metric_reports(&self) -> Vec<LoadMetricReport> {
+        self
+            .primary_load_metric_reports
+            .iter()
```

Comment:
Just return the PrimaryLoadMetricReportList. make the field private.
mssf api prefers not use Vec or other types that requires dynamic alloc for list.